metis-repository/src/main/java/eu/europeana/metis/repository/dao/RecordDao.java

```diff
@@ -0,0 +1,73 @@
+package eu.europeana.metis.repository.dao;
+
+import com.mongodb.client.MongoClient;
+import dev.morphia.Datastore;
+import dev.morphia.Morphia;
+import dev.morphia.mapping.DiscriminatorFunction;
+import dev.morphia.mapping.MapperOptions;
+import dev.morphia.mapping.NamingStrategy;
+import dev.morphia.query.experimental.filters.Filters;
+import eu.europeana.metis.network.ExternalRequestUtil;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Database Access Object for {@link Record} instances.
+ */
+public class RecordDao {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RecordDao.class);
+  private static final String RECORD_ID_FIELD = "recordId";
+  private static final String DATASET_ID_FIELD = "datasetId";
+
+  private final Datastore datastore;
+
+  /**
+   * Constructor.
+   *
+   * @param mongo Client to the mongo database.
+   * @param databaseName The name of the database.
+   */
+  public RecordDao(MongoClient mongo, String databaseName) {
+    final MapperOptions mapperOptions = MapperOptions.builder().discriminatorKey("className")
+        .discriminator(DiscriminatorFunction.className())
+        .collectionNaming(NamingStrategy.identity()).build();
+    this.datastore = Morphia.createDatastore(mongo, databaseName, mapperOptions);
+    this.datastore.getMapper().map(Record.class);
+  }
+
+  /**
+   * Create record in the database
+   *
+   * @param record - The record to be saved in the database
+   * @return The record just saved in the database
+   */
+  public Record createRecord(Record record) {
+
+    Optional<Record> recordFound = datastore.find(Record.class)
+        .filter(Filters.eq(RECORD_ID_FIELD, record.getRecordId())).stream().findFirst();
+
+    recordFound.ifPresent(value -> record.setId(value.getId()));
+
+    final Record recordSaved = ExternalRequestUtil.retryableExternalRequestForNetworkExceptions(
+        () -> datastore.save(record));
+    LOGGER.info("Record for datasetId '{}' created in Mongo", record.getDatasetId());
+
+    return recordSaved;
+  }
+
+  /**
+   * Creates  list of records that belong to the given dataset id
+   *
+   * @param datasetId - The id of the dataset the records belong to
+   * @return A List of records that are part of the dataset with datasetId
+   */
+  public List<Record> getAllRecordsFromDataset(String datasetId) {
+    return datastore.find(Record.class).filter(Filters.eq(DATASET_ID_FIELD, datasetId)).stream()
+        .collect(Collectors.toList());
+  }
```

Comment:
I think it is better to return a stream here, instead of loading everything in memory first.

Reply:
Also works, I just changed the method